src/rgw/driver/rados/rgw_d3n_datacache.h

```diff
@@ -41,20 +41,21 @@ struct D3nChunkDataInfo : public LRUObject {
 
 struct D3nCacheAioWriteRequest {
 	std::string oid;
-	void *data;
-	int fd;
-	struct aiocb *cb;
-	D3nDataCache *priv_data;
-	CephContext *cct;
+	void *data = nullptr;
+	int fd = -1;
+	struct aiocb *cb = nullptr;
+	D3nDataCache *priv_data = nullptr;
+	CephContext *cct = nullptr;
 
 	D3nCacheAioWriteRequest(CephContext *_cct) : cct(_cct) {}
 	int d3n_prepare_libaio_write_op(bufferlist& bl, unsigned int len, std::string oid, std::string cache_location);
 
   ~D3nCacheAioWriteRequest() {
-    ::close(fd);
-		cb->aio_buf = nullptr;
-		free(data);
-		data = nullptr;
+    if (fd >= 0) {
```

Comment:
if (fd >= 0) should not be needed, and imo, should not be trusted.  in particular, when would fd == 0;  when it's STDIN_FILENO?  similarly when fd == 1.

Reply:
what I'm suggesting is, iff we did the open(), we should unconditionally close